src/ChunkMap.h

```diff
@@ -491,28 +431,27 @@ class cChunkMap
 			LOG("Out of Memory");
 		}
 	};
-	
-	typedef std::list<cChunkLayer *> cChunkLayerList;
-	
-	typedef std::list<cChunkStay *> cChunkStays;
 
-	/** Finds the cChunkLayer object responsible for the specified chunk; returns nullptr if not found. Assumes m_CSLayers is locked. */
-	cChunkLayer * FindLayerForChunk(int a_ChunkX, int a_ChunkZ);
-	
-	/** Returns the specified cChunkLayer object; returns nullptr if not found. Assumes m_CSLayers is locked. */
-	cChunkLayer * FindLayer(int a_LayerX, int a_LayerZ);
-	
-	/** Returns the cChunkLayer object responsible for the specified chunk; creates it if not found. */
-	cChunkLayer * GetLayerForChunk (int a_ChunkX, int a_ChunkZ);
-	
-	/** Returns the specified cChunkLayer object; creates it if not found. */
-	cChunkLayer * GetLayer(int a_LayerX, int a_LayerZ);
+	struct ChunkCoordinate
+	{
+		bool operator() (const ChunkCoordinate & a_Lhs, const ChunkCoordinate & a_Rhs) const
+		{
+			return ((a_Lhs.ChunkX == a_Rhs.ChunkX) ? (a_Lhs.ChunkZ < a_Rhs.ChunkZ) : (a_Lhs.ChunkX < a_Rhs.ChunkX));
+		}
+
+		int ChunkX;
+		int ChunkZ;
+	};
 	
-	void RemoveLayer(cChunkLayer * a_Layer);
+	typedef std::list<cChunkStay *> cChunkStays;
 
 	cCriticalSection m_CSLayers;
-	cChunkLayerList  m_Layers;
-	cEvent           m_evtChunkValid;  // Set whenever any chunk becomes valid, via ChunkValidated()
+
+	/** A map of chunk coordinates to chunk pointers
+	Uses a map (as opposed to unordered_map) because sorted maps are apparently faster */
+	std::map<ChunkCoordinate, std::unique_ptr<cChunk>, ChunkCoordinate> m_Chunks;
```

Comment:
How big was the world used to generate the statistics? Because std::map generally involves a tree, with O(log N growth)